packages/sandpack-core/src/resolver/resolver.ts

```diff
@@ -184,12 +200,17 @@ function* resolveNodeModule(
 
       try {
         const pkgFilePath = pathUtils.join(rootDir, pkgSpecifierParts.filepath);
-        const pkgJson = yield* loadPackageJSON(pkgFilePath, opts, rootDir);
+        const rootPkgJson = yield* loadPackageJSON(rootDir, opts);
+        const pkgJson =
+          rootPkgJson && rootPkgJson.content.hasExports
+            ? rootPkgJson
+            : yield* loadNearestPackageJSON(pkgFilePath, opts, rootDir);
```

Comment:
Isn't this duplicate of rootPkgJson, when would this not be rootPkgJson?

Reply:
I mention this in [the other comment](https://github.com/codesandbox/codesandbox-client/pull/7380/files#r1095856464) - "nested" `package.json` files are possible and actually participate in the "classic" resolution algorithm. So it's still required to read from them - but only if the root one doesn't have `exports`

Reply:
Makes sense, thanks for explaining